registry_provider_version.go

```diff
@@ -0,0 +1,267 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ RegistryProviderVersions = (*registryProviderVersions)(nil)
+
+// RegistryProviderVersions describes the registry provider version methods that
+// the Terraform Enterprise API supports.
+//
+// TFE API docs: https://www.terraform.io/cloud-docs/api-docs/private-registry/provider-versions-platforms
+type RegistryProviderVersions interface {
+	// List all versions for a single provider.
+	List(ctx context.Context, providerID RegistryProviderID, options *RegistryProviderVersionListOptions) (*RegistryProviderVersionList, error)
+
+	// Create a registry provider version.
+	Create(ctx context.Context, providerID RegistryProviderID, options RegistryProviderVersionCreateOptions) (*RegistryProviderVersion, error)
+
+	// Read a registry provider version.
+	Read(ctx context.Context, versionID RegistryProviderVersionID) (*RegistryProviderVersion, error)
+
+	// Delete a registry provider version.
+	Delete(ctx context.Context, versionID RegistryProviderVersionID) error
+}
+
+// registryProvidersVersions implements RegistryProvidersVersions
+type registryProviderVersions struct {
+	client *Client
+}
+
+// RegistryProviderVersion represents a registry provider version
+type RegistryProviderVersion struct {
+	ID                 string                             `jsonapi:"primary,registry-provider-versions"`
+	Version            string                             `jsonapi:"attr,version"`
+	CreatedAt          string                             `jsonapi:"attr,created-at,iso8601"`
+	UpdatedAt          string                             `jsonapi:"attr,updated-at,iso8601"`
+	KeyID              string                             `jsonapi:"attr,key-id"`
+	Protocols          []string                           `jsonapi:"attr,protocols"`
+	Permissions        RegistryProviderVersionPermissions `jsonapi:"attr,permissions"`
+	ShasumsUploaded    bool                               `jsonapi:"attr,shasums-uploaded"`
+	ShasumsSigUploaded bool                               `jsonapi:"attr,sasums-sig-uploaded"`
+
+	// Relations
+	RegistryProvider          *RegistryProvider           `jsonapi:"relation,registry-provider"`
+	RegistryProviderPlatforms []*RegistryProviderPlatform `jsonapi:"relation,platforms"`
+
+	// Links
+	Links map[string]interface{} `jsonapi:"links,omitempty"`
+}
+
+// RegistryProviderVersionID is the multi key ID for addressing a version provider
+type RegistryProviderVersionID struct {
+	RegistryProviderID
+	Version string
+}
+
+type RegistryProviderVersionPermissions struct {
+	CanUploadAsset bool `jsonapi:"attr,can-upload-asset,omitempty"`
```

Comment:
This can't be omitted because bool can't be empty. Or, perhaps false values would be considered empty? Either way, it's undesired

Reply:
It looks like can_delete is also a field in this [response](https://github.com/hashicorp/atlas/blob/main/app/serializers/api/v2/registry_provider_version_serializer.rb#L66-L69)?

Reply:
Correct, the only way you can omit a bool is to make the value a pointer.

Reply:
That's a good point. And in reviewing the [docs](https://app.asana.com/0/1202314304798782/1202383446952123/f) for registry provider version, there are two permissions that are serialized. I'll update this.